### PR TITLE
Added textFieldProps to SelectField in uniforms-material (closes #700).

### LINKED ISSUE #700: Unable to propagate 'classes' prop for SelectField on uniforms-material
I am trying to style the FormControl in a SelectField, but neither style or classes are being provided to the TextField component.

I believe that at least the classes prop should be explicitly provided to TextField: https://github.com/vazco/uniforms/blob/master/packages/uniforms-material/src/SelectField.tsx

### PR DESCRIPTION
**Pull request by Monteth**
This PR introduces `textFieldProps` for `SelectField`. It allows setting `TextField` properties as its the kinda `wrapper` for `Select` (in contrast to `checkboxes`, where `wrapField` holds inner components). 

Also, I removed  `defaultProps` from all components in `uniforms-material`.

### CODE REVIEW

File: packages/uniforms-material/src/DateField.tsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ function Date({
   disabled,
   error,
   errorMessage,
+  fullWidth = true,
```
Comment (radekmie) [inline packages/uniforms-material/src/DateField.tsx:26]: You forgot to remove `Date.defaultProps` though.

File: packages/uniforms-material/src/LongTextField.tsx [hunk 2]
```diff
@@ -1,26 +1,29 @@
-import React, { ReactNode } from 'react';
+import React from 'react';
 import TextField, { StandardTextFieldProps } from '@material-ui/core/TextField';
-import { FieldProps, connectField, filterDOMProps } from 'uniforms';
+import { connectField, FieldProps, filterDOMProps } from 'uniforms';
```
Comment (radekmie) [inline packages/uniforms-material/src/LongTextField.tsx:3]: Is your IDE configured to do that? I _still_ didn't managed to configure that, but please, keep them ordered.

File: packages/uniforms-material/src/NestField.tsx [hunk 3]
```diff
@@ -1,26 +1,29 @@
 import FormLabel from '@material-ui/core/FormLabel';
-import React, { HTMLProps } from 'react';
-import { HTMLFieldProps, connectField } from 'uniforms';
+import React from 'react';
+import { connectField, HTMLFieldProps } from 'uniforms';
 
 import AutoField from './AutoField';
 import wrapField from './wrapField';
 
+// FIXME: wrapField is not typed correctly.
```
Comment (radekmie) [inline packages/uniforms-material/src/NestField.tsx:8]: Maybe it's a good idea to do it now?

File: packages/uniforms-material/src/NumField.tsx [hunk 4]
```diff
@@ -28,11 +28,13 @@ function Num({
 }: NumFieldProps) {
   return (
     <TextField
-      disabled={!!disabled}
+      disabled={disabled}
       error={!!error}
+      fullWidth={props.fullWidth ?? true}
```
Comment (radekmie) [inline packages/uniforms-material/src/NumField.tsx:33]: I'd rather use `fullWidth = true,` in the arguments lists as we do that in other components.

File: packages/uniforms-material/src/SelectField.tsx [hunk 5]
```diff
@@ -122,31 +118,27 @@ function Select(props: SelectFieldProps) {
       {
         ...props,
         component: 'fieldset',
-        disabled,
-        error,
-        errorMessage,
-        showInlineError,
       },
```
Comment (radekmie) [inline packages/uniforms-material/src/SelectField.tsx:121]: I suggest simplifying it to:

```ts
{ ...props, component: 'fieldset' },
```

**Review by radekmie — CHANGES_REQUESTED** [review]

File: packages/uniforms-material/src/LongTextField.tsx (hunk 2 above)
Comment (Monteth) [reply packages/uniforms-material/src/LongTextField.tsx:3]: Yeah, I fixed 4 of 5 exactly the same rearrangements in other files, this one escaped me.

File: packages/uniforms-material/src/SelectField.tsx [hunk 6]
```diff
@@ -162,7 +150,8 @@ function Select(props: SelectFieldProps) {
 
   return (
     <TextField
-      disabled={!!disabled}
+      {...filterDOMProps(props)}
```
Comment (radekmie) [inline packages/uniforms-material/src/SelectField.tsx:153]: It looks like we cannot do that - there's `...filterDOMProps(props)` in `SelectProps` in line 172.

**Review by radekmie — COMMENTED** [review]
Looks good! Let's just add the documentation for new props.

**Review by radekmie — APPROVED** [review]

**Review by kestarumper — APPROVED** [review]

**Pull request merged**